src/frontends/tensorflow/src/decoder_proto.cpp

```diff
@@ -285,16 +285,53 @@ size_t DecoderProto::get_input_size() const {
     return m_node_def->input_size();
 }
 
-void DecoderProto::get_input_node(size_t input_port_idx,
-                                  std::string& producer_name,
-                                  size_t& producer_output_port_index) const {
-    // Body graph nodes may have two colons `:`, for example,
-    // producer_name:z:2 means that producer operation name is `producer_name`
-    // and output port is 2
-    std::string producer_port_name = m_node_def->input(static_cast<int>(input_port_idx));
+void parse_producer_name(const std::string& producer_port_name,
+                         std::string& producer_name,
+                         size_t& producer_output_port_index,
+                         const DecoderBase::OpTypeByName& op_type_by_name) {
+    using OutputPortIdxMax = std::unordered_map<std::string, int>;
+    // create a table of operation type and its output ports
+    // for which we specify output port indices manually
+    // it is mainly affects multiple output operations
+    // extract this information from tensorflow/core/ops/*.cc files
+    const OutputPortIdxMax output_port_idx_map = {
+        {"TopK:indices", 1},
+        {"TopKV2:indices", 1},
+        {"CTCGreedyDecoder:decoded_values", 1},
+        {"CTCGreedyDecoder:decoded_shape", 2},
+        {"CTCGreedyDecoder:log_probability", 3},
+        {"CTCGreedyDecoder:log_probability", 3},
+        {"FusedBatchNorm:batch_mean", 1},
+        {"FusedBatchNorm:batch_variance", 2},
+        {"FusedBatchNormV2:batch_mean", 1},
+        {"FusedBatchNormV2:batch_variance", 2},
+        {"FusedBatchNormV3:batch_mean", 1},
+        {"FusedBatchNormV3:batch_variance", 2},
+    };
+    // Body graph nodes may have two colons `:` input names, for example,
+    // `TopKV2Name:indices:0` means that producer operation name is `TopKV2Name`
+    // the middle name is output port name of the producer `indices` that means
+    // the second output port of TopKV2 is used.
+    // The first output port of TopKV2 is described as `TopKV2Name:values:0`
     auto first_colon = producer_port_name.find_first_of(":");
     auto last_colon = producer_port_name.find_last_of(":");
-    if (first_colon != std::string::npos && last_colon != std::string::npos) {
+    if (first_colon != std::string::npos && first_colon < last_colon) {
+        // we have at least two colons producer_name:output_port_name:port_idx
+        producer_name = producer_port_name.substr(0, first_colon);
+        auto port_id = producer_port_name.substr(last_colon + 1);
+        auto port_name = producer_port_name.substr(first_colon + 1, last_colon - first_colon - 1);
+        FRONT_END_GENERAL_CHECK(!port_id.empty() && std::all_of(port_id.begin(), port_id.end(), ::isdigit),
```

Comment:
Why it may contain only digits? Please, refer on documentation it could only decimal indeces.

Reply:
described above in the comment